src/overlays/actors/ovl_En_Dns/z_en_dns.c

```diff
@@ -74,44 +73,34 @@ static ColliderCylinderInitType1 sCylinderInit = {
     { 18, 32, 0, { 0, 0, 0 } },
 };
 
-static u16 D_809F040C[] = {
+static u16 sItemDialogRef[] = {
     0x10A0, 0x10A1, 0x10A2, 0x10CA, 0x10CB, 0x10CC, 0x10CD, 0x10CE, 0x10CF, 0x10DC, 0x10DD,
 };
 
 // Debug text: "sells"  { "Deku Nuts",    "Deku Sticks",        "Piece of Heart",  "Deku Seeds",
 //                        "Deku Shield",  "Bombs",              "Arrows",          "Red Potion",
 //                        "Green Potion", "Deku Stick Upgrade", "Deku Nut Upgrade" }
-static char* D_809F0424[] = {
+static char* sItemDebugTxt[] = {
     "デクの実売り            ", "デクの棒売り            ", "ハートの欠片売り        ", "デクの種売り            ",
     "デクの盾売り            ", "バクダン売り            ", "矢売り                  ", "赤のくすり売り          ",
     "緑のくすり売り          ", "デクの棒持てる数を増やす", "デクの実持てる数を増やす",
 };
 
-static DnsItemEntry D_809F0450 = { 20, 5, GI_DEKU_NUTS_5_2, func_809EF5A4, func_809EFA28 };
-
-static DnsItemEntry D_809F0460 = { 15, 1, GI_DEKU_STICKS_1, func_809EF658, func_809EF9F8 };
-
-static DnsItemEntry D_809F0470 = { 10, 1, GI_HEART_PIECE, func_809EF70C, func_809EFA58 };
-
-static DnsItemEntry D_809F0480 = { 40, 30, GI_DEKU_SEEDS_30, func_809EF73C, func_809EF9F8 };
-
-static DnsItemEntry D_809F0490 = { 50, 1, GI_SHIELD_DEKU, func_809EF800, func_809EF9F8 };
-
-static DnsItemEntry D_809F04A0 = { 40, 5, GI_BOMBS_5, func_809EF854, func_809EFA9C };
-
-static DnsItemEntry D_809F04B0 = { 70, 20, GI_ARROWS_30, func_809EF8F4, func_809EFACC };
-
-static DnsItemEntry D_809F04C0 = { 40, 1, GI_BOTTLE_POTION_RED, func_809EF9A4, func_809EF9F8 };
-
-static DnsItemEntry D_809F04D0 = { 40, 1, GI_BOTTLE_POTION_GREEN, func_809EF9A4, func_809EF9F8 };
-
-static DnsItemEntry D_809F04E0 = { 40, 1, GI_DEKU_STICK_UPGRADE_20, func_809EF70C, func_809EFAFC };
-
-static DnsItemEntry D_809F04F0 = { 40, 1, GI_DEKU_NUT_UPGRADE_30, func_809EF70C, func_809EFB40 };
+static DnsItemEntry sItemNuts = { 20, 5, GI_DEKU_NUTS_5_2, EnDns_ItemNutsDialog, EnDns_ItemsNutsPay };
+static DnsItemEntry sItemSticks = { 15, 1, GI_DEKU_STICKS_1, EnDns_ItemsSticksDialog, EnDns_ItemsPricePay };
```

Comment:
don't the "dialog" functions look more like they're checking for prerequisites to buying the item? like, having slingshot, enough space, enough rupees
with "dialog" I'd expect some textId stuff

Reply:
Agreed. In en_ossan and [en_girl_a](https://github.com/zeldaret/oot/blob/9f09505d34619883748a7dab05071883281c14fd/src/overlays/actors/ovl_En_GirlA/z_en_girla.c#L156) I named these functions `CanBuy`, and they return a `CANBUY` enum. https://github.com/zeldaret/oot/blob/9f09505d34619883748a7dab05071883281c14fd/src/overlays/actors/ovl_En_GirlA/z_en_girla.h#L92-L99

I feel it would be sensible give the two more consistent names, though unfortunately EnDnsDialogType and EnGirlACanBuyResult appear to be different enumerations